src/windows/export.py

```diff
@@ -468,77 +485,83 @@ def cboSimpleTarget_index_changed(self, widget, index):
+                            vc = xmldoc.getElementsByTagName("videocodec")
+                            self.txtVideoCodec.setText(vc[0].childNodes[0].data)
+                            sr = xmldoc.getElementsByTagName("samplerate")
+                            self.txtSampleRate.setValue(int(sr[0].childNodes[0].data))
+                            c = xmldoc.getElementsByTagName("audiochannels")
+                            self.txtChannels.setValue(int(c[0].childNodes[0].data))
+                            c = xmldoc.getElementsByTagName("audiochannellayout")
+
+                            # check for compatible audio codec
+                            ac = xmldoc.getElementsByTagName("audiocodec")
+                            audio_codec_name = ac[0].childNodes[0].data
+                            if audio_codec_name == "aac":
+                                # Determine which version of AAC encoder is available
+                                if openshot.FFmpegWriter.IsValidCodec("libfaac"):
+                                    self.txtAudioCodec.setText("libfaac")
+                                elif openshot.FFmpegWriter.IsValidCodec("libvo_aacenc"):
+                                    self.txtAudioCodec.setText("libvo_aacenc")
+                                elif openshot.FFmpegWriter.IsValidCodec("aac"):
+                                    self.txtAudioCodec.setText("aac")
+                                else:
+                                    # fallback audio codec
+                                    self.txtAudioCodec.setText("ac3")
                             else:
                                 # fallback audio codec
-                                self.txtAudioCodec.setText("ac3")
-                        else:
-                            # fallback audio codec
-                            self.txtAudioCodec.setText(audio_codec_name)
-
-                        layout_index = 0
-                        for layout in self.channel_layout_choices:
-                            if layout == int(c[0].childNodes[0].data):
-                                self.cboChannelLayout.setCurrentIndex(layout_index)
-                                break
-                            layout_index += 1
-
+                                self.txtAudioCodec.setText(audio_codec_name)
+
+                            layout_index = 0
+                            for layout in self.channel_layout_choices:
+                                if layout == int(c[0].childNodes[0].data):
+                                    self.cboChannelLayout.setCurrentIndex(layout_index)
+                                    break
+                                layout_index += 1
+                                
+                    except ExpatError as e:
+                        # This indicates an invalid Preset file - display an error and continue
+                        log.error("Failed to parse file '%s' as a preset: %s" % (preset_path, e))
```

Comment:
This won't actually _display_ an error anywhere in the console output — which means only in the logfile, for Windows users, since they don't even get console output. (Ditto macOS and Linux users who launch OpenShot graphically, but at least they have the option.) That's probably a _good_ thing, most of the time — we don't really want to be throwing error dialogs up at users because of a profile parsing issue. There are already spots in the code that show dialog messages for errors loading _icons_, something I've argued against because what is the user expected to do about that!?

But in this case, especially if it's one of the user's own profiles, it might be good to give them some indication that there's an issue, because that's something they **can** correct. So, it might be worth showing an unobtrusive error message (localized, also) in the status bar of the main window. It's a sadly-underutilized means of communicating information without interrupting the user or forcing them to deal with the message immediately. You can see it used in the main window's code for handling the save frame / snapshot preview feature, though.

Reply:
Oh, that's right, and also in the Recent Project list handling. This bit here:
https://github.com/OpenShot/openshot-qt/blob/a22fa5db739bd1c70a9c16912658522e6fc5468e/src/windows/main_window.py#L537-L540

Reply:
(There's code in `__init__`,  I believe, that sets up `self.statusBar` for the class.)